CMakeLists.txt

```
@@ -215,9 +230,20 @@ set(OPENCV_LIBS
     "libopencv_calib3d.so"
     "libopencv_flann.so"
     "libopencv_features2d.so"
-    "libpng16.so"
-    "libjpeg.so"
 )
+if(${RHEL_BUILD})
+    set(OPENCV_LIBS
+        ${OPENCV_LIBS}
+        "libjpeg.so.62"  # RHEL uses only .so.62 version
+        "libpng16.so.16" # RHEL uses .so.16 version
+    )
+else()
+    set(OPENCV_LIBS
+        ${OPENCV_LIBS}
+        "libjpeg.so"
+        "libpng16.so"
+    )
+endif()
```

Using [generator-expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html) is more preferable in my opinion: 
```suggestion
  set(OPENCV_LIBS
         ${OPENCV_LIBS}
         $<IF:$<BOOL:RHEL_BUILD>,libjpeg.so.62,libjpeg.so>
         $<IF:$<BOOL:RHEL_BUILD>,libpng16.so.16,libpng16.so>
  )
```
see: https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#conditional-expressions

---

From my perspective, using if/else is more readable and evaluates during the configuration phase. However, I just noticed that we're adopting [generator expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html) across other backends outside of PyTorch. If we want to maintain uniformity in our approach, I can make the changes.

---

Feel free to ignore my suggestions and resolve conversations as needed.